### PR TITLE
Fix parsing long flag

### LINKED ISSUE #54: Incorrect parsing for long param flag with dashes
First example:
```nu
#!/usr/bin/env nu

def main [
  --long-param-with-dashes
  --long_param_with_underscores
] {
  if $long_param_with_dashes {
    print "dash param"
  } else {
    print "no dash param"
  }

  if $long_param_with_underscores {
    print "underscore param"
  } else {
    print "no underscore param"
  }
}
```
Syntax highlighting is incorrect and there are multiple "param_short_flag" for the first param flag (with dashes):
tree-sitter-nu1

Second example (probably same cause but different effect):
```nu
#!/usr/bin/env nu

def main [
  --dry-run (-n)
] {
  if $dry_run {
    print "Dry run"
  } else {
    print "Real execution"
  }
}
```
There is now an error in the tree:
tree-sitter-nu2

### PR DESCRIPTION
**Pull request by kit494way**
Fix #54

### CODE REVIEW

File: corpus/decl/def.nu [hunk 1]
```diff
@@ -110,3 +110,199 @@ def-env test []: nothing -> record<a: string, b: int> {}
...
+        (param_long_flag
+          (identifier))))
+    (block)))
+
+======
+def-008-long-flag-with-short
+======
+
+def test [
+  --long-flag (-l)
```
Comment (fdncred) [inline corpus/decl/def.nu:138]: does this require a space or will it parse like `--long-flag(-l)`?

**Comment by fdncred** [comment]
Looks good but what about this type of flags?

```nushell
def test [
  --long-flag(-l): int
] {}
```

File: corpus/decl/def.nu (hunk 1 above)
Comment (kit494way) [reply corpus/decl/def.nu:138]: A space is not required.

**Comment by kit494way** [comment]
I will add more tests.

**Comment by kit494way** [comment]
I added some tests.
I also fixed a same bug in a rule for long flags in command arguments.

**Comment by fdncred** [comment]
thanks! i'm really enjoying all these fixes/additions. keep up the great work!

**Pull request merged**